src/Controller/ExportController.php

```diff
@@ -289,9 +294,20 @@ private function buildXmlProducts(
     ): array {
         $items = [];
 
+        $crossSellingCategories = $this->getConfig(
+            'crossSellingCategories',
+            $salesChannelContext->getSalesChannel()->getId()
+        );
+
         /** @var ProductEntity $productEntity */
         foreach ($productEntities as $productEntity) {
             try {
+                $categories = $productEntity->getCategories();
+                $category = $categories ? $categories->first() : null;
+                $categoryId = $category ? $category->getId() : null;
+                if (!empty($crossSellingCategories) && in_array($categoryId, $crossSellingCategories, false)) {
+                    throw new ProductHasCrossSellingCategoryException($categoryId);
```

Comment:
Throw the full message including the cross sell category, which caused that.

```
Product with id <ID> (<product name>) was not exported because it is assigned to cross selling category <ID> (<category name>)
```

This way it is easier to solve it when reading the logs. If it is easy possible, please get the category tree, this way you can just see the log and know what to do.